dynamic_programming/longest_increasing_subsequence.py

```diff
@@ -1,12 +1,36 @@
-"""
-The Longest Increasing Subsequence (LIS) problem is to find the length of the longest subsequence of a given sequence such that all elements of the subsequence are sorted in increasing order. For example, the length of LIS for {10, 22, 9, 33, 21, 50, 41, 60, 80} is 6 
-"""
-def LIS(arr):
-    n= len(arr)
-    lis = [1]*n
-
-    for i in range(1, n):
-        for j in range(0, i):
-            if arr[i] > arr[j] and lis[i] <= lis[j]:
-                lis[i] = lis[j] + 1
-    return max(lis)
+'''
+Author  : Mehdi ALAOUI
+
+This is a pure Python implementation of Dynamic Programming solution to the longest increasing subsequence of a given sequence.
+
+The problem is  :
+Given an ARRAY, to find the longest and increasing sub ARRAY in that given ARRAY and return it.
+Example: [10, 22, 9, 33, 21, 50, 41, 60, 80] as input will return [10, 22, 33, 41, 60, 80] as output
+'''
```

Comment:
I just finished some optimizations on complexity, it was, on the worst case O(2^(n-1)), but now, I think that in the worst case, it's O(nlog(n)), if I'm wrong, please correct me.